graphql_compiler/tests/test_input_data.py

```diff
@@ -37,6 +37,47 @@ def immediate_output():
         type_equivalence_hints=None)
 
 
+def immediate_output_custom_scalars():
+    graphql_input = '''{
+        Animal {
+            birthday @output(out_name: "birthday")
+            net_worth @output(out_name: "net_worth")
+        }
+    }'''
+    expected_output_metadata = {
+        'birthday': OutputMetadata(type=GraphQLDate, optional=False),
+        'net_worth': OutputMetadata(type=GraphQLDecimal, optional=False),
+    }
+    expected_input_metadata = {}
+
+    return CommonTestData(
+        graphql_input=graphql_input,
+        expected_output_metadata=expected_output_metadata,
+        expected_input_metadata=expected_input_metadata,
+        type_equivalence_hints=None)
+
+
+def immediate_output_with_custom_scalar_filter():
+    graphql_input = '''{
+        Animal {
+            name @output(out_name: "animal_name")
+            net_worth @filter(op_name: ">=", value: ["$min_worth"])
```

Comment:
Mostly irrelevant: ?aybe add a birthday filter here as well (to parallel the previous test)? Of course date filters are tested elsewhere too. So, not really necessary

Reply:
I would prefer not to, since having two filters in the same query would implicitly test other code (e.g. the lowering steps that merge consecutive filters). It's nice to keep this test small and limited to as little code as possible, so we can pinpoint issues more easily.